airflow/config_templates/default_celery.py

```diff
@@ -23,13 +23,27 @@
 from airflow.exceptions import AirflowConfigException, AirflowException
 from airflow.utils.log.logging_mixin import LoggingMixin
 
+
+def _broker_supports_visibility_timeout(url):
+    # Only certain transports support the 'visibility_timeout' option.
+    #
+    # See:
+    # http://docs.celeryproject.org/en/master/userguide/configuration.html#broker-transport-options
+    if url.startswith("redis") or url.startswith("sqs"):
+        return True
+    return False
+
+
 log = LoggingMixin().log
 
+broker_url = configuration.conf.get('celery', 'BROKER_URL')
+
 broker_transport_options = configuration.conf.getsection(
     'celery_broker_transport_options'
 )
-if broker_transport_options is None:
-    broker_transport_options = {'visibility_timeout': 21600}
+if 'visibility_timeout' not in broker_transport_options:
+    if _broker_supports_visibility_timeout(broker_url):
```

Comment:
Oh wait. This isn't quite right.

This will always set it for redis/sqs if it's not there, but before it would only set it if the config was empty. This is a slight change in behaviour that is probably unintended?

Reply:
`broker_transport_options` was never None, because `[celery_broker_transport_options]` was added to default_airflow.cfg.  However `visibility_timeout` was hardcoded in default_airflow.cfg.  But `visibility_timeout` is incompatible with backends other than redis or sqs.

If I commented out `visibility_timeout` from default_airflow.cfg, then `broker_transport_options` would still never be None...it would just be an empty dictionary.

So, to keep the existing behavior of always setting visibility_timeout for redis/sqs if it was not set in airflow.cfg, but at the same time not breaking sqla, I chose the approach in the patch.

Reply:
Ah, so the `is None` was never true. I see :)

Yes this approach seems good then.